src/moi_oracle.jl

```diff
@@ -53,6 +53,222 @@ function compute_extreme_point(
+            # VariableIndex GreaterThan-constraint is already set, needs to be deleted first
+            if is_set
+                c_idx = var_constraint_list[func]
+                MOI.delete(o, c_idx)
+            end
+            MOI.add_constraint(o, func, MOI.EqualTo(set.upper))
+            set_equal = true
+        end
+    elseif S <: MOI.Interval
+        if set.upper ≈ val || set.lower ≈ val
+            set_equal = true
+            if set.upper ≈ val
+                MOI.add_constraint(o, func, MOI.EqualTo(set.upper))
+            else
+                MOI.add_constraint(o, func, MOI.EqualTo(set.lower))
+            end
+        end
+    end
+    if !set_equal 
+        idx = MOI.add_constraint(o, func, set)
+        var_constraint_list[func] = idx
+    end     
+end
+
+function compute_inface_extreme_point!(lmo::MathOptLMO{OT}, direction, x; kwargs...) where {OT}
+    var_constraint_list = Dict([])
+    lmo2 = copy(lmo)
+    MOI.empty!(lmo2.o)
+    MOI.set(lmo2.o, MOI.Silent(), true)
+    variables = MOI.get(lmo.o, MOI.ListOfVariableIndices())
+    terms = [MOI.ScalarAffineTerm(d, v) for (d, v) in zip(direction, variables)]
+    obj = MOI.ScalarAffineFunction(terms, zero(Float64))
+    MOI.set(lmo2.o, MOI.ObjectiveFunction{typeof(obj)}(), obj)
+    for (F, S) in MOI.get(lmo.o, MOI.ListOfConstraintTypesPresent())
+        valvar(f) = x[f.value]
+        const_list = MOI.get(lmo.o, MOI.ListOfConstraintIndices{F,S}())
+        for c_idx in const_list
+            func = MOI.get(lmo.o, MOI.ConstraintFunction(), c_idx)
+            val = MOIU.eval_variables(valvar, func)
+            set = MOI.get(lmo.o, MOI.ConstraintSet(), c_idx)
+            set_constraint(lmo2.o, S, func, val, set, var_constraint_list)  
+        end
+    end
+    MOI.set(lmo2.o, MOI.ObjectiveSense(), MOI.MIN_SENSE)
+    MOI.optimize!(lmo2.o)
+    a = MOI.get(lmo2.o, MOI.VariablePrimal(), variables)
+    MOI.empty!(lmo2.o)
+    return a
+end
+
+# Second version of compute_inface_extreme_point.
+# Copy and modify the constriants if necesssary.
+function compute_inface_extreme_point(lmo::MathOptLMO{OT}, direction, x;solve_data=Dict(), kwargs...) where {OT}
+    lmo2 = copy(lmo)
+    MOI.set(lmo2.o, MOI.Silent(), true)
+    variables = MOI.get(lmo2.o, MOI.ListOfVariableIndices())
+    MOI.add_variables(lmo2.o, length(variables))
+    terms = [MOI.ScalarAffineTerm(d, v) for (d, v) in zip(direction, variables)]
+    obj = MOI.ScalarAffineFunction(terms, zero(Float64))
+    MOI.set(lmo2.o, MOI.ObjectiveFunction{typeof(obj)}(), obj)
+    for (F, S) in MOI.get(lmo2.o, MOI.ListOfConstraintTypesPresent())
+        valvar(f) = x[f.value]
+        const_list = MOI.get(lmo2.o, MOI.ListOfConstraintIndices{F,S}())
+        for c_idx in const_list
+            func = MOI.get(lmo2.o, MOI.ConstraintFunction(), c_idx)
+            val = MOIU.eval_variables(valvar, func)
+            set = MOI.get(lmo2.o, MOI.ConstraintSet(), c_idx)
+            if S <: MOI.GreaterThan
```

Comment:
For this, it's better to use a function barrier to reduce the amount of type instability